hbase-rsgroup/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupAdminEndpoint.java

```diff
@@ -291,17 +292,19 @@ public void removeRSGroup(RpcController controller,
     @Override
     public void balanceRSGroup(RpcController controller,
         BalanceRSGroupRequest request, RpcCallback<BalanceRSGroupResponse> done) {
+      BalanceRequest balanceRequest = RSGroupProtobufUtil.toBalanceRequest(request);
+
       BalanceRSGroupResponse.Builder builder = BalanceRSGroupResponse.newBuilder();
       LOG.info(master.getClientIdAuditPrefix() + " balance rsgroup, group="
           + request.getRSGroupName());
       try {
-        if (master.getMasterCoprocessorHost() != null) {
+        if (master.getMasterCoprocessorHost() != null && !balanceRequest.isDryRun()) {
```

Comment:
Why skip the coprocessor call-back? Shouldn't the coprocessor be informed with the full content of the request object?

Reply:
The idea here is that the dryRun mode should have no impact on the cluster, and we can't really guarantee that with a coprocessor. I could add the dry run argument to the coprocessor, but I imagine that opens another can of interface worms. I'm fine to add it, I just felt it might cause more issues than not. Thoughts?

Reply:
As coprocessors go, I think it's not up HBase to decide what behavior a coprocessor implements, only to faithfully delegate calls at the defined extension points. I think the dry_run mode should pass through any installed coprocessors, which means this change will require a change to coprocessor APIs. This falls more explicitly under the terms of HBase LimitedPrivate API, which means it will require a minor version bump. But it's a change to the public API, so I assume a minor version bump was required anyway.

If you're concerned that making this change is too meddlesome for this patch, I'm okay with a follow-on jira being filed and a TODO being left in the code here.

Reply:
Ok sounds good. I'll look into doing it here and if it's too much of a change we can go the TODO route.

Yea I think minor version is fine. I'm going to backport this to our fork once merged anyway.

Reply:
It ended up being pretty easy to modify the interface, so I did that here since we're already in the "requires minor version" territory.